mobile/src/modules/price-api/store/actions.ts

```diff
@@ -18,12 +19,26 @@ export const loadPriceApiData = createActionFn<void, Promise<void>>(
   }
 );
 
+export const selectCurrency = createActionFn<PriceTypeStrings, Promise<void>>(
+  async (dispatch, _getState, currency) => {
+    dispatch(actions.selectCurrency(currency));
+  }
+);
+
 export const loadHistoricalPriceApiData = createActionFn<void, Promise<void>>(
   async (dispatch, _getState) => {
-    const response = await fetch(defaultSettings.cryptoCompareURL);
-    const updatedPriceInfo = await response.json();
-    if (updatedPriceInfo.Data && updatedPriceInfo.Data.length) {
-      dispatch(actions.updateHistoricalPriceInfo(updatedPriceInfo));
+    const response = await Promise.all([
```

Comment:
I'm more into `loadHistoricalApiData(cur1); loadHistoricalApiData(cur2);` cuz if second one will fail for some reason, then we still have first one. In case `Promise.all` we won't get anything if something went wrong.
And it will be easier to add another currencies if needed

Reply:
yeah I want to improve error handling and add loading indicators too, just not yet.

Reply:
also look at modules/transactions/store for errors, loading and so on